service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt

```diff
@@ -78,120 +89,163 @@ private fun Database.Read.getUnitOperationalDays(unitId: DaycareId, dateRange: F
-        FROM daycare JOIN daycare_group ON daycare.id = daycare_group.daycare_id
-        WHERE daycare.id = :unitId
-        UNION ALL
-        SELECT
-            daycare.id AS unit_id,
-            daycare.name AS unit_name,
-            NULL AS group_id,
-            NULL AS group_name
-        FROM daycare
-        WHERE daycare.id = :unitId
-    ), group_res AS (
-        SELECT
-            child.id,
-            child.first_name,
-            child.last_name,
-            child.date_of_birth,
-            res.start_time,
-            res.end_time,
-            res.start_date,
-            group_placement.daycare_group_id AS group_id
-        FROM person child
-        JOIN placement ON child.id = placement.child_id AND daterange(placement.start_date, placement.end_date, '[]') && :dateRange
-        LEFT JOIN attendance_reservation res ON child.id = res.child_id AND res.start_date BETWEEN placement.start_date AND placement.end_date
-        LEFT JOIN daycare_group_placement group_placement ON placement.id = group_placement.daycare_placement_id
-            AND daterange(placement.start_date, placement.end_date, '[]') && daterange(group_placement.start_date, group_placement.end_date, '[]')
-    )
-    SELECT
-        unit_group.unit_name AS unit,
-        unit_group.group_name AS group,
-        group_res.id,
-        group_res.first_name,
-        group_res.last_name,
-        group_res.date_of_birth,
-        group_res.start_date AS date,
-        to_char((group_res.start_time AT TIME ZONE 'Europe/Helsinki')::time, 'HH24:MI') AS start_time,
-        to_char((group_res.end_time AT TIME ZONE 'Europe/Helsinki')::time, 'HH24:MI') AS end_time
-    FROM unit_group
-    LEFT JOIN group_res ON unit_group.group_id = group_res.group_id OR (unit_group.group_id IS NULL AND group_res.group_id IS NULL)
+    SELECT 
+        t::date AS date,
+        dg.name AS "group",
+        p.id,
+        p.first_name,
+        p.last_name,
+        p.date_of_birth,
+        to_char((res.start_time AT TIME ZONE 'Europe/Helsinki')::time, 'HH24:MI') AS reservation_start_time,
+        to_char((res.end_time AT TIME ZONE 'Europe/Helsinki')::time, 'HH24:MI') AS reservation_end_time,
+        att.arrived AS attendance_start,
+        att.departed AS attendance_end,
+        ab.absence_type
+    FROM generate_series(:start, :end, '1 day') t
+    JOIN placement pl ON pl.unit_id = :unitId AND daterange(pl.start_date, pl.end_date, '[]') @> t::date
+    JOIN person p ON p.id = pl.child_id
+    LEFT JOIN daycare_group_placement dgp on dgp.daycare_placement_id = pl.id AND daterange(dgp.start_date, dgp.end_date, '[]') @> t::date
+    LEFT JOIN daycare_group dg ON dg.id = dgp.daycare_group_id
+    LEFT JOIN attendance_reservation res ON res.child_id = p.id AND res.start_date = t::date
+    LEFT JOIN child_attendance att ON att.child_id = p.id AND (att.arrived AT TIME ZONE 'Europe/Helsinki')::date = t::date
+    LEFT JOIN absence ab ON ab.child_id = p.id AND ab.date = t::date
```

Comment:
This join means that multiple rows are returned when a child has multiple absences during the same day.

Reply:
Right.. any suggestion on how to fix? I guess on DB level they may even have different absence types, but the UI spec does not support showing multiple reasons for absence per day.

Reply:
I don't really know how it should be interpreted in case the absence types differ, but I would just take a random one with a lateral join for now.